src/runtime-rs/crates/hypervisor/src/device/driver/vfio.rs

```diff
@@ -456,7 +457,12 @@ impl Device for VfioDevice {
             .await
             .context("failed to increase attach count")?
         {
-            return Err(anyhow!("attach count increased failed as some reason."));
+            warn!(
+                sl!(),
+                "It's not allowed that device {:?} is attached many times,", self.device_id
```

Comment:
I think it is more concise.

```suggestion
                "The device {:?} is not allowed to be attached multiple times.", self.device_id
```